docs/admin/performance_tuning.md

```diff
@@ -1,229 +1,165 @@
 
-### Building CaRT self_test
+The mode is selected via the `--master-endpoint` option. If this option is
+notified on the command line, then we are in the first mode and the self_test
+binary itself issues the RPCs. If one or several master endpoint are specified,
+then we are in the cross-server mode.
 
-The CaRT `self_test` and its tests are delivered as part of the daos_client
-and daos_tests [distribution packages][2]. It can also be built from scratch.
+An endpoint consists of pair of two values separated by a colon. The first value
+is the rank that matches the engine rank displayed in `dmg system query`. The
+second value is called tag and identified the service thread in the engine.
+The DAOS engine uses the following mapping:
 
-```bash
-$ git clone --recurse-submodules https://github.com/daos-stack/daos.git
-$ cd daos
-$ scons --build-deps=yes install
-$ cd install
-```
+- tag 0 is used by the metadata service handling pool and container operations.
+- tag 1 is used for cross-server monitoring (SWIM).
+- tags 2 to [#targets + 1] is used by DAOS targets (one tag per target].
+- tags [#targets + 1] to [#targets + #helpers] is used by helper service threads.
 
-For detailed information, please refer to the [DAOS build documentation][3]
-section.
+As an example, an engine with `targets: 16` and `nr_xs_helpers: 4` would use the
+following tag distributions:
+- tag 0: metadata service
+- tag 1: monitoring service
+- tag 2-17: targets 0 to 15 (16 targets total)
+- tag 17-20: helper service
 
-### Running CaRT self_test
+For a total of 21 endpoints exported by this engine.
 
-Instructions to run CaRT `self_test` are as follows.
+The RPC flow sent over the network can be configured via the `--message-sizes`
+options that take a list of size tuples. Performance will be reported individually
+for each tuple. Each size integer can be prepended with a single character to specify
+the underlying transport mechanism. Available types are:
 
-**Start DAOS server**
+- 'e' - Empty (no payload)
+- 'i' - I/O vector (IOV)
+- 'b' - Bulk transfer
 
-`self_test` requires DAOS server to be running before attempt running
-`self_test`. For detailed instruction on how to start DAOS server, please refer
-to the [server startup][4] documentation.
+For example, (b1000) would transfer 1000 bytes via bulk in both directions. Similarly,
+(i100 b1000) would use IOV to send and bulk to reply.
 
-**Dump system attachinfo**
+`--repetitions-per-size` can be used to define the Number of samples per message
+size per endpoint with a default value of 10000. `--max-inflight-rpcs`
+determines the number of concurrent RPCs issued simultaneously.
 
-`self_test` will use the address information in `daos_server.attach_info_tmp`
-file. To create such file, run the following command:
+`self_test` has many more options. For a full description of `self_test` usage,
+please run:
 
 ```bash
-./bin/daos_agent dump-attachinfo -o ./daos_server.attach_info_tmp
+$ ./bin/self_test --help
```

Comment:
drop "./bin/" ? should be in the PATh for ROM install, and we don't say what ".". is...

Reply:
fixed.